command-line/lib/Options/Applicative/Script.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Options.Applicative.Script
+    (
+    -- ** Applicative Parser
+      scriptArg
+
+    -- * Internal
+    , requireSignatureOfParser
+    , requireAllOfParser
+    , requireAnyOfParser
+    , requireAtLeastOfParser
+    , scriptParser
+
+    ) where
+
+import Prelude
+
+import Cardano.Multisig
+    ( Script (..), keyHashFromBytes, validateScript )
+import Codec.Binary.Encoding
+    ( fromBase16 )
+import Data.Char
+    ( isDigit, isLetter )
+import Data.Word
+    ( Word8 )
+import Options.Applicative
+    ( Parser, argument, eitherReader, help, metavar )
+import Text.ParserCombinators.ReadP
+    ( ReadP, readP_to_S, (<++) )
+
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Text.ParserCombinators.ReadP as P
+
+
+scriptFromString :: String -> Either String Script
+scriptFromString str =
+    case readP_to_S scriptParser str of
+         [(multisig,_rest)] ->
+             if (validateScript multisig) then
+                 Right multisig
+             else
+                 Left "The script is invalid."
+         _ ->
+             Left "Parsing of the script failed."
```

Comment:
Error reporting is perhaps one area where a better parsing library would help. But never mind, this script format is so simple it probably doesn't matter.

Reply:
good point. I was wondering how to expose upstream errors from parser ... probably there is a way anyway. Agree that it is some profit to stick to ReadP as it does not impose new dependencies...and the script format is very simple